agent/consul/leader_connect_ca.go

```diff
@@ -909,6 +909,9 @@ func (c *CAManager) UpdateConfiguration(args *structs.CARequest) (reterr error)
 		if respErr, ok := resp.(error); ok {
 			return respErr
 		}
+		if respOk, ok := resp.(bool); ok && !respOk {
```

Comment:
I think in practice this code path will not be hit because the FSM operation will return an error. Even during an upgrade this code probably won't be relevant because the leader handles both operations.

I think unnecessary code like this can be a distraction and can lead to other problems, so I suggest we do not add this handling.

Reply:
The issue is that the FSM code is error prone, the fact that we return a flag and an error and the flag is not really used can lead to other errors as this one. 
Another possibility would be to return always false or true so it's explicit that the response is not used for that specific operation

Reply:
I think the `bool` return value is the problem. The error should be able to indicate if it was success or not. The bool is a distraction that we should either remove or ignore.

Reply:
I will remove this from here and create another PR to refactor it to be less error prone.